kuksa_databroker/databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -54,55 +55,156 @@ impl proto::val_server::Val for broker::DataBroker {
             let mut errors = Vec::new();
 
             for request in requested {
-                match broker.get_entry_by_path(&request.path).await {
-                    Ok(entry) => {
-                        let view = proto::View::from_i32(request.view).ok_or_else(|| {
-                            tonic::Status::invalid_argument(format!(
-                                "Invalid View (id: {}",
-                                request.view
-                            ))
-                        })?;
-                        let fields =
-                            HashSet::<proto::Field>::from_iter(request.fields.iter().filter_map(
-                                |id| proto::Field::from_i32(*id), // Ignore unknown fields for now
-                            ));
-                        let fields = combine_view_and_fields(view, fields);
-                        debug!("Getting fields: {:?}", fields);
-                        let proto_entry = proto_entry_from_entry_and_fields(entry, fields);
-                        debug!("Getting datapoint: {:?}", proto_entry);
-                        entries.push(proto_entry);
-                    }
-                    Err(ReadError::NotFound) => {
-                        errors.push(proto::DataEntryError {
-                            path: request.path,
-                            error: Some(proto::Error {
-                                code: 404,
-                                reason: "not_found".to_owned(),
-                                message: "Path not found".to_owned(),
-                            }),
-                        });
-                    }
-                    Err(ReadError::PermissionExpired) => {
-                        errors.push(proto::DataEntryError {
-                            path: request.path,
-                            error: Some(proto::Error {
-                                code: 401,
-                                reason: "unauthorized".to_owned(),
-                                message: "Authorization expired".to_owned(),
-                            }),
-                        });
+                match glob::is_valid_pattern(&request.path) {
+                    true => {
```

Comment:
It's probably better to "short circuit" if the path is invalid, to avoid at least one level of nesting. I.e.

```rust
for request in requested {
    if glob::is_valid_pattern(&request.path) {
        errors.push(proto::DataEntryError {
            path: request.path,
            error: Some(proto::Error {
                code: 400,
                reason: "bad_request".to_owned(),
                message: "Bad Path Request".to_owned(),
            }),
        });
        continue;
    }
```

(and as a side note, matching on `true` / `false` is not idiomatic Rust as that's what `if` / `else` are for... But you quickly get used to using `match` for everything in Rust :) )

Reply:
The update still nests everything in a big if / else. Still think it would be better to short circuit at the top of the `for request in requested` loop.

It's probably better to "short circuit" if the path is invalid, to avoid at least one level of nesting. I.e.

```rust
for request in requested {
    if !glob::is_valid_pattern(&request.path) {
        errors.push(proto::DataEntryError {
            path: request.path,
            error: Some(proto::Error {
                code: 400,
                reason: "bad_request".to_owned(),
                message: "Bad Path Request".to_owned(),
            }),
        });
        continue;
    }

    let view = ...;
    let fields = ...;
    match broker.get_entry_by_path(&request.path).await {
```

Reply:
Done